mvpa2/measures/label.py

```diff
@@ -0,0 +1,139 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the PyMVPA package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Label adjacent features in a dataset
+"""
+import numpy as np
+from collections import defaultdict
+from operator import itemgetter
+
+from ..base.param import Parameter
+from ..base.constraints import EnsureBool
+from ..datasets.base import Dataset
+from ..misc.neighborhood import CachedQueryEngine
+
+from .base import Measure
+
+if __debug__:
+    from ..base import debug
+
+
+class ClustersLabeler(Measure):
+    """Determines contiguous regions of features using a query engine.
+
+    "label" in the name chosen to mimic scipy.ndimage.measurements.label
+    """
+    qe = Parameter(
+        None,
+        doc="QueryEngine to determine neighbors of any given feature")
```

Comment:
Will the cluster computation depend on the "radius" of the query engine? Or would it be possible to extract a neighborhood structure that takes only the closest neighboring samples?